File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for(int i=1; i<brown; i++){
            for(int j=1; j<brown; j++){
                if((i*j == brown+yellow) && (j>=i)
                  &&((j-2)*(i-2) == yellow)){
                    answer[0] = j;
                    answer[1] = i;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    
    w, h = 3, 3
    
    for i in range(3, brown+1):
        for j in range(3, brown+1):
            if j>=i and (j-2)*(i-2)==yellow and j*i == brown+yellow:
                w, h = j, i
    answer = [w, h]
    return answer